tools/deployment/k8s/s3/deploy.sh

```diff
@@ -0,0 +1,130 @@
+else
+    RELEASE_TARBALL_URL=$(curl --silent --show-error \
+      "https://api.github.com/repos/y-scope/yscope-log-viewer/releases/${TAG_NAME}" | \
+      jq -r '.assets[0].browser_download_url')
+fi
+if [ -z "$RELEASE_TARBALL_URL" ]; then
+  LOG "[ERROR] Could not automatically resolve release tarball URL"
+  exit 1
+fi
+
+# Wait until S3 endpoint is available
+LOG "[INFO] Waiting until s3://${LOG_VIEWER_BUCKET} endpoint becomes available ..."
+wait_for_s3_availability
+
+# Create log-viewer bucket if not already exist
+LOG "[INFO] Creating s3://${LOG_VIEWER_BUCKET} bucket"
+if ! aws s3api head-bucket --endpoint-url "$AWS_ENDPOINT_URL" --bucket "$LOG_VIEWER_BUCKET" 2>/dev/null; then
+    aws s3api create-bucket --endpoint-url "$AWS_ENDPOINT_URL" --bucket "$LOG_VIEWER_BUCKET"
+else
+    LOG "[WARN] Bucket s3://${LOG_VIEWER_BUCKET} already exists"
+fi
+
+# Define and apply the Bucket Policy for public read access
+LOG "[INFO] Applying public read access policy to s3://${LOG_VIEWER_BUCKET}"
+POLICY=$(cat <<EOP
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Principal": "*",
+            "Action": "s3:GetObject",
+            "Resource": "arn:aws:s3:::${LOG_VIEWER_BUCKET}/*"
+        }
+    ]
+}
+EOP
+)
+if ! aws s3api put-bucket-policy \
+  --endpoint-url "${AWS_ENDPOINT_URL}" --bucket "${LOG_VIEWER_BUCKET}" --policy "$POLICY"; then
+    LOG "[ERROR] Failed to set bucket policy for s3://${LOG_VIEWER_BUCKET}"
+    exit 1
+fi
+
+# Download and decompress release tarball to a temp directory
+LOG "[INFO] Downloading ${RELEASE_TARBALL_URL}"
+DECOMPRESSED_ASSETS_DIRECTORY="${DECOMPRESSED_ASSETS_DIRECTORY:-/tmp/yscope-log-viewer}"
+mkdir -p "$DECOMPRESSED_ASSETS_DIRECTORY"
+if ! curl --silent --show-error --location "$RELEASE_TARBALL_URL" | \
+  tar --strip-components 1 -xz -C "$DECOMPRESSED_ASSETS_DIRECTORY"; then
+    LOG "[ERROR] Failed to download and extract the release tarball from ${RELEASE_TARBALL_URL}"
+    exit 1
+fi
+
+# Upload all assets to object store at the root of the provided bucket
+# Note that uploads can fail with invalid/unknown checksum sent error.
+# This typically occurs with old MinIO. If this happens, update to release after late 2024.
+# See this GitHub issue for details: https://github.com/minio/minio/pull/19680
+LOG "[INFO] Uploading yscope-log-viewer assets to object store"
+aws s3 cp "$DECOMPRESSED_ASSETS_DIRECTORY" "s3://${LOG_VIEWER_BUCKET}/" \
+    --recursive --endpoint-url "$AWS_ENDPOINT_URL"
+
+LOG "[INFO] Deployment completed successfully"
+
+# Print out a prompt message with a box around it
+prompt_message="yscope-log-viewer is now available at ${AWS_ENDPOINT_URL}/${LOG_VIEWER_BUCKET}/index.html"
+message_length=${#prompt_message}
+total_length=$((message_length + 6))
+printf "\n\n"
+printf "+%${total_length}s+\n" | tr ' ' "-"
+printf "|%${total_length}s|\n"
+printf "|   $prompt_message   |\n"
```

Comment:
```suggestion
printf "|   %s   |\n" "$prompt_message"
```

To address [SC2059](https://www.shellcheck.net/wiki/SC2059).